tests/arch_tests.cmake

```diff
@@ -154,6 +154,56 @@ function(add_xc7_test)
     add_custom_target(${arch}-${test_name}-bit DEPENDS ${bit})
     add_dependencies(all-tests ${arch}-${test_name}-bit)
     add_dependencies(all-${device}-tests ${arch}-${test_name}-bit)
+
+    # generate vivado timing report
+    set(vivado_report ${output_dir}/vivado_report.txt)
+    set(vivado_timing_tcl ${CMAKE_SOURCE_DIR}/tests/common/timing_dump_vivado.tcl)
+    message(STATUS "${arch} ${test_name} ${board} ${name}")
+    add_custom_command(
+        OUTPUT ${vivado_report}
+        COMMAND ${CMAKE_COMMAND} -E env
+            VIVADO_SETTINGS=${VIVADO_SETTINGS}
+            ${quiet_cmd}
+            ${run_vivado} -mode tcl -source ${vivado_timing_tcl} -tclargs ${dcp} ${vivado_report}
+        DEPENDS
+            ${dcp}
+    )
+
+    add_custom_target(${arch}-${test_name}-vivado-report DEPENDS ${vivado_report})
+
+    # generate custom timing report
+    set(custom_report ${output_dir}/custom_report.txt)
+    set(phys ${output_dir}/${name}.phys)
+    set(device_db ${CMAKE_BINARY_DIR}/devices/${device}/${device}_timing.device)
+    add_custom_command(
+        OUTPUT ${custom_report}
+        COMMAND ${CMAKE_COMMAND} -E env
+            python3 -m fpga_interchange.static_timing_analysis --schema_dir ${INTERCHANGE_SCHEMA_PATH} --device ${device_db}
+            --physical_netlist ${phys} --compact > ${custom_report}
```

Comment:
python3 should be taken from the `programs` target: https://github.com/SymbiFlow/fpga-interchange-tests/blob/84c7ea5711ccef16062e9b70da00e41c80d8bc3e/tests/tests.cmake#L67.

Also it might be more readable to have each argument in a new line

Reply:
Fixed and reformated